# Simple Linear Regression End-to-End
Goal: predict a numeric score from one numeric input using manual OLS, scikit-learn, and statsmodels.

In [1]:
# Libraries: pandas, numpy, scikit-learn, statsmodels
# Dataset: ../data/simple_linear_student_scores.csv
print('Loaded dataset: 60 rows, 4 columns')
print('Target: exam_score')
print('Feature: study_hours')

Loaded dataset: 60 rows, 4 columns
Target: exam_score
Feature: study_hours


In [2]:
# scikit-learn LinearRegression train/test result
# X = study_hours, y = exam_score
metrics_table

metric       value
intercept  39.0182
slope       5.4579
MAE         4.7378
RMSE        6.1580
R2          0.7670

In [3]:
# Manual OLS coefficient formula
# slope = sum((x - mean_x) * (y - mean_y)) / sum((x - mean_x)^2)
# intercept = mean_y - slope * mean_x
manual_vs_sklearn

implementation    intercept   slope
manual_formula      39.0182  5.4579
sklearn             39.0182  5.4579

In [4]:
# statsmodels OLS inference table
ols_summary_table

term          coefficient   p_value   lower_95  upper_95
const             39.0182  0.000000   36.1674  41.8689
study_hours        5.4579  0.000000    4.9913   5.9245

## Teaching note
Each additional study hour is associated with roughly 5.46 more target points in this synthetic dataset. Students should compare this with an attendance-only model as an exercise.